status-im/status-desktop
# feat: channels encryption backend

### What does the PR do

This brings channels encryption backend. Now every token-gated channel should be encrypted behind the scenes. It also includes improved community rekeying.

### Affected areas

- communities messaging
    - on encrypted community
    - on encrypted channels
    
### Test scenarios

#### Test community-level encryption
1) Create token-gated community
1) Join with MemberA who satisfies the criteria
1) Send message "X" to general channel
1) MemberA should see the "X" message
1) Change token permissions so that MemberA no longer satisfies the criteria
1) MemberA should be kicked from the community
1) Send message "Y" to general channel
1) Change token permissions so that MemberA once again meets the criteria
1) MemberA should be re-added to the community
1) MemberA should see both "X" and "Y" messages
1) Send message "Z" to general channel
1) MemberA should see "X", "Y" and "Z" messages

#### Test channel-level encryption
1) Create open community
1) Create token-gated test channel
1) Join with MemberA who satisfies test channel's criteria
1) Send message "X" to general and test channels
1) MemberA should see the "X" message both in general and test channels
1) Change test channel's token permissions so that MemberA no longer satisfies the criteria
1) Send message "Y" to general and test channels
1) Member should see the "X" and "Y" messages in general channel but not in the test channel
1) Change token permissions so that MemberA once again meets the criteria
1) MemberA should see both "X" and "Y" messages in the test channel
1) Send message "Z" to general and test channels
1) MemberA should see "X", "Y" and "Z" messages in both the general and test channels

@glitchminer please consider adding above scenarios to regressions tests
